src/tools/build_helper/src/git.rs

```diff
@@ -28,3 +31,40 @@ pub fn get_rust_lang_rust_remote() -> Result<String, String> {
         rust_lang_remote.split('.').nth(1).ok_or_else(|| "remote name not found".to_owned())?;
     Ok(remote_name.into())
 }
+
+pub fn rev_exists(rev: &str, git_dir: Option<&Path>) -> Result<bool, String> {
+    let mut git = Command::new("git");
+    if let Some(git_dir) = git_dir {
+        git.current_dir(git_dir);
+    }
+    git.args(["rev-parse", rev]);
+    let output = git.output().map_err(|err| format!("{err:?}"))?;
+
+    match output.status.code() {
+        Some(0) => Ok(true),
+        Some(128) => Ok(false),
+        None => Ok(false), // we don't know, let's assume it doesn't
```

Comment:
this should also return an error, I think - if code() returns none that means git was interrupted by a signal, maybe a segfault or something.